examples/ablation-workshop-mpi.py

```diff
@@ -0,0 +1,763 @@
+
+        Function specific for verification against the ablation workshop test
+        case 2.1.
+        """
+        cv = state.cv
+        dv = state.dv
+
+        wdv = gas_model.wall.dependent_vars(dv.wall_density)
+
+        actx = cv.mass.array_context
+
+        # restrict variables to the domain boundary
+        dd_vol_quad = dd_wall.with_discr_tag(quadrature_tag)
+        bdtag = dd_wall.trace("prescribed").domain_tag
+        normal_vec = actx.thaw(dcoll.normal(bdtag))
+        dd_bdry_quad = dd_vol_quad.with_domain_tag(bdtag)
+
+        temperature_bc = op.project(dcoll, dd_wall, dd_bdry_quad, dv.temperature)
+        m_dot_g = op.project(dcoll, dd_wall, dd_bdry_quad, cv.mass*velocity)
+        emissivity = op.project(dcoll, dd_wall, dd_bdry_quad, wdv.emissivity)
+
+        m_dot_g = np.dot(m_dot_g, normal_vec)
+
+        # time-dependent function
+        weight = actx.np.where(actx.np.less(time, 0.1), (time/0.1)+1e-7, 1.0)
+
+        h_e = 1.5e6*weight
+        conv_coeff_0 = 0.3*weight
+
+        # this is only valid for the non-ablative case (2.1)
+        m_dot_c = 0.0
+        m_dot = m_dot_g + m_dot_c + 1e-13
+
+        # ~~~~
+        # blowing correction: few iterations to converge the coefficient
+        conv_coeff = conv_coeff_0*1.0
+        lambda_corr = 0.5
+        for _ in range(0, 3):
+            phi = 2.0*lambda_corr*m_dot/conv_coeff
+            blowing_correction = phi/(actx.np.exp(phi) - 1.0)
+            conv_coeff = conv_coeff_0*blowing_correction
+
+        Bsurf = m_dot_g/conv_coeff  # noqa N806
+
+        # ~~~~
+        # get the wall enthalpy using spline interpolation
+        bnds_T = bprime_class._bounds_T  # noqa N806
+        bnds_B = bprime_class._bounds_B  # noqa N806
+
+        # couldn't make lazy work without this
+        sys.setrecursionlimit(10000)
+
+        # using spline for temperature interpolation
+        # while using "nearest neighbor" for the "B" parameter
+        h_w = 0.0
+        for j in range(0, 24):
+            for k in range(0, 15):
+                h_w = \
+                    actx.np.where(actx.np.greater_equal(temperature_bc, bnds_T[j]),
+                    actx.np.where(actx.np.less(temperature_bc, bnds_T[j+1]),
+                    actx.np.where(actx.np.greater_equal(Bsurf, bnds_B[k]),
+                    actx.np.where(actx.np.less(Bsurf, bnds_B[k+1]),
+                          bprime_class._cs_Hw[k, 0, j]*(temperature_bc-bnds_T[j])**3
+                        + bprime_class._cs_Hw[k, 1, j]*(temperature_bc-bnds_T[j])**2
+                        + bprime_class._cs_Hw[k, 2, j]*(temperature_bc-bnds_T[j])
+                        + bprime_class._cs_Hw[k, 3, j], 0.0), 0.0), 0.0), 0.0) + h_w
+
+        h_g = gas_model.eos.gas_enthalpy(temperature_bc)
+
+        flux = -(conv_coeff*(h_e - h_w) - m_dot*h_w + m_dot_g*h_g)
+
+        radiation = emissivity*5.67e-8*(temperature_bc**4 - 300**4)
+
+        return make_obj_array([flux - radiation])
```

Comment:
Changed in [efa834b](https://github.com/illinois-ceesd/mirgecom/pull/875/commits/efa834b1505237ddead31d538b704996717109a3)

Apparently, it was necessary because of the flux@normal product.